src/transformers/modeling_utils.py

```diff
@@ -1512,6 +1513,40 @@ def _autoset_attn_implementation(
             # If a config is passed with a preset attn_implementation, we skip the automatic dispatch and use the user-provided config, with hard checks that the requested attention implementation is available.
             requested_attn_implementation = config._attn_implementation_internal
 
+        # MultiModal LLM related hack: since they consist of two or might be more sub-models
+        # we have to check and dispatch SDPA to each sub-model, in case any of them support it.
+        # If one sub-model supports SDPA while other doesn't, an error will be raised following the
+        # typical SDPA-dispatch path.
+        # Same goes for the `_supports_cache_class` because we cannot know what flag LM has
+        # before knowing which class is that LM
```

Comment:
Yes, so that if one sub-model can dispatch on SDPA, it will dispatch. But the other will not. At the same time the general `config._attn_implementation` flag in for ex. LLaVa will be "sdpa" , and will not be used while init